openc3/lib/openc3/io/posix_serial_driver.rb

```diff
@@ -132,9 +137,19 @@ def read
       begin
         data = @handle.read_nonblock(65535)
       rescue Errno::EAGAIN, Errno::EWOULDBLOCK
-        result = IO.fast_select([@handle], nil, nil, @read_timeout)
-        if result
-          retry
+        begin
+          read_ready, _ = IO.fast_select(@readers, nil, nil, @read_timeout)
+        rescue IOError
+          @pipe_reader.close unless @pipe_reader.closed?
```

Comment:
I don't see how closing the pipe_reader is going to help with the serial port handle

Reply:
Closing the pipe reader is just to make sure we don't leave file handles open.   Return "" means that we are dead and should no longer be used going forward.  IOError should only occur if the serial port has already been closed.